### PR TITLE
fix(stoneintg-753): requeue release creation until timeout

### PR DESCRIPTION
**Pull request by 14rcole**
Prior to this commit the integration service would requeue release creation indefinitely if it failed.  If a release quota was hit then releases would be backed up then be created all at once and overwhelm user workspaces with potentially no-longer-relevant releases.  This change sets a timeout of three hours for requeued releases, after which creating a release for that snapshot will fail.

## Maintainers will complete the following section

- [x] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [x] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [x] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

**Review by MartinBasti — CHANGES_REQUESTED** [review]
LGTM except this small issue with not using constants

File: controllers/snapshot/snapshot_adapter.go [hunk 1]
```diff
@@ -934,3 +935,14 @@ func GetDetailsFromStatusError(err error) (string, string) {
 	}
 	return errKind, errName
 }
+
```
Comment (jsztuka) [inline controllers/snapshot/snapshot_adapter.go:938]: Do we want to add docstring here?

Comment (14rcole) [reply controllers/snapshot/snapshot_adapter.go:938]: Yes, thank you

File: controllers/snapshot/snapshot_adapter.go [hunk 2]
```diff
@@ -934,3 +935,14 @@ func GetDetailsFromStatusError(err error) (string, string) {
...
+func (a *Adapter) RequeueIfYoungerThanThreshold(retErr error) (controller.OperationResult, error) {
+	timeout := time.Duration(3 * time.Hour)
```
[the remark this answers is not in the document]
Comment (MartinBasti) [reply controllers/snapshot/snapshot_adapter.go]: sorry I mean it as a module constant, not just const. Something that could be exported into unittest

File: controllers/snapshot/snapshot_adapter.go [hunk 3]
```diff
@@ -435,9 +436,9 @@ func (a *Adapter) EnsureAllReleasesExist() (controller.OperationResult, error) {
 		if er != nil {
 			a.logger.Error(er, "Failed to mark snapshot integration status as invalid",
 				"snapshot.Name", a.snapshot.Name)
-			return controller.RequeueWithError(errors.Join(err, er))
+			return a.RequeueIfYoungerThanThreshold(errors.Join(err, er))
```
Comment (14rcole) [inline controllers/snapshot/snapshot_adapter.go:439]: It seems like this line and the other three like it are the cause of the "Duplicated Lines" error in Sonarcloud.  Is there any way to get around this? It doesn't look like I can combine the lines in any way.

Comment (MartinBasti) [reply controllers/snapshot/snapshot_adapter.go:439]: I'd just ignore it

File: controllers/snapshot/snapshot_adapter.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (14rcole) [reply controllers/snapshot/snapshot_adapter.go]: Got it, thanks for the clarification

**Review by MartinBasti — APPROVED** [review]

**Review by Josh-Everett — APPROVED** [review]

**Pull request merged**